### PR TITLE
[Refactoring] Use try with resources to properly close resources

### PR DESCRIPTION
**Pull request by Shadow-Devil**
## Purpose
Many resources in Java need be closed after they have been used. If they are not, the garbage collector cannot reclaim the resources' memory, and they are still considered to be in use by the operating system. Such resources are considered to be leaked, which can lead to performance issues.

Java 7 introduced the try-with-resources statement, which guarantees that the resource in question will be closed.
```java
try (InputStream input = Files.newInputStream(path)) {
  // "input" will be closed after the execution of this block
}
```
This syntax is safer than the traditional method using `try`, `catch`, and `finally` and hence should be preferred.

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: ballerina-shell/modules/shell-cli/src/main/java/io/ballerina/shell/cli/jline/JlineTerminalAdapter.java
Comment (hasithaa) [inline ballerina-shell/modules/shell-cli/src/main/java/io/ballerina/shell/cli/jline/JlineTerminalAdapter.java:62]: As a practice, we are not using `var` keyword. Use the Class name here.

Comment (Shadow-Devil) [reply ballerina-shell/modules/shell-cli/src/main/java/io/ballerina/shell/cli/jline/JlineTerminalAdapter.java:62]: @hasithaa 
Does that also apply to cases where the variable is declared inside a try with resource?
Example:
```java
try (var paths = Files.walk(src)) {
    paths.forEach(source -> copy(source, dest.resolve(src.relativize(source))));
}
```

**Review by SasinduDilshara — APPROVED** [review]

**Comment by Shadow-Devil** [comment]
Remove stale status

**Comment by Shadow-Devil** [comment]
Fixed merge conflicts via rebase

File: tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/util/HttpClientRequest.java
Comment (gimantha) [inline tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/util/HttpClientRequest.java:119]: We can include writer as well into the try block. Will send the changes separately

File: cli/central-client/src/main/java/org/ballerinalang/central/client/Utils.java
Comment (gimantha) [inline cli/central-client/src/main/java/org/ballerinalang/central/client/Utils.java:468]: This can be merged with the outer try block. I will send those changes in a sepaerta PR

File: misc/lib-creator/src/main/java/org/ballerinalang/stdlib/utils/BuildLangLib.java
Comment (gimantha) [inline misc/lib-creator/src/main/java/org/ballerinalang/stdlib/utils/BuildLangLib.java:118]: We can insert the all these statements into the try block

**Comment by Shadow-Devil** [comment]
@gimantha thank you for the review. Can you try commiting and pushing your suggestions to this branch? I enabled write access for maintainers of the project to the branch. I think this is easier than creating a sepsrate PR or waiting for me to implement your feedback :)

**Comment by gimantha** [comment]
> @gimantha thank you for the review. Can you try commiting and pushing your suggestions to this branch? I enabled write access for maintainers of the project to the branch. I think this is easier than creating a sepsrate PR or waiting for me to implement your feedback :)

Sure.. I tried it failed. I will try again

**Comment by gimantha** [comment]
@Shadow-Devil for some reason, the upstream build are failing with my update. So I reverted it. I will  push to a different branch and see if the build passes or not.

**Comment by Shadow-Devil** [comment]
@gimantha Hm, that is weird. At least now the build of the original PR is successful (only codecov is failing because of code coverage).

**Comment by gimantha** [comment]
> @gimantha Hm, that is weird. At least now the build of the original PR is successful (only codecov is failing because of code coverage).

Yes.. I will check that separately and go ahead and merge this PR. Thank you for the contribution @Shadow-Devil

**Review by gimantha — APPROVED** [review]

**Pull request merged**
